frontend/dockerfile/instructions/bflag.go

```diff
@@ -147,25 +147,32 @@ func (bf *BFlags) Parse() error {
 		return errors.Wrap(bf.Err, "error setting up flags")
 	}
 
-	for _, arg := range bf.Args {
-		if !strings.HasPrefix(arg, "--") {
-			return errors.Errorf("arg should start with -- : %s", arg)
-		}
-
-		if arg == "--" {
+	for _, a := range bf.Args {
+		if a == "--" {
+			// Stop processing further arguments as flags. We're matching
+			// the POSIX Utility Syntax Guidelines here;
+			// https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap12.html#tag_12_02
+			//
+			// > The first -- argument that is not an option-argument should be accepted
+			// > as a delimiter indicating the end of options. Any following arguments
+			// > should be treated as operands, even if they begin with the '-' character.
 			return nil
 		}
+		if !strings.HasPrefix(a, "--") {
+			return errors.Errorf("arg should start with -- : %s", a)
+		}
 
-		arg, value, hasValue := strings.Cut(arg[2:], "=")
+		flagName, value, hasValue := strings.Cut(a, "=")
+		arg := a[2:]
```

Comment:
^ this line seems to be the bug

Reply:
DOH!!! I see what I did there; was originally taking a slightly different approach 🙈 . Let me fix that